src/librustc_mir/borrow_check/mod.rs

```diff
@@ -373,10 +373,14 @@ fn do_mir_borrowck<'a, 'gcx, 'tcx>(
                     for err in &mut mbcx.errors_buffer {
                         if err.is_error() {
                             err.level = Level::Warning;
-                            err.warn("This error has been downgraded to a warning \
-                                      for backwards compatibility with previous releases.\n\
-                                      It represents potential unsoundness in your code.\n\
-                                      This warning will become a hard error in the future.");
+                            err.warn(
+                                "this error has been downgraded to a warning for backwards \
+                                 compatibility with previous releases",
```

Comment:
I would restructure the warning as follows:

> this error has been downgraded to a warning, for backwards compatibility with previous releases, and will become a hard error in the future.
> this represents potential undefined behavior in your code.

It feels more direct; I also swapped out unsoundness for undefined behavior since that's a more commonly known term I think.

Reply:
I don't know if I prefer that message. I tried to keep the message the way it was and just make it more consistent with other diagnostics. 

Restructing the message in that way would make the first warning quite long - I normally try to keep diagnostics short so they are unlikely to get line wrapped (and handle that manually by emitting more than one warning/note as I've done here). I also think that the second warning could be interpreted as being unrelated to the first warning. I don't have any preference on "unsoundness" vs "undefined behaviour". 

I don't feel too strongly about the exact message though - as long as the casing/wrapping is made consistent.

Reply:
@davidtwco how about if you just swap in "undefined behavior" in place of "unsoundness" then, and leave this otherwise as it is?